### ISSUE TITLE
Failing integration test on travis for inabox friendly iframe

### ISSUE THREAD

**Issue by lannka** [ampproject/amphtml#20319]
https://travis-ci.org/ampproject/amphtml/jobs/479586023

**Comment by lannka** [comment]
@zombifier could you take a look?

**Comment by zombifier** [comment]
#20318 can be merged if we want to make master green again for the time being. I suspect it may be due to the host script failing to be loaded (port 8000 vs port 8081).

**Comment by zombifier** [comment]
Alright, I think I know why the test fails. Travis on PRs run `gulp build`, which builds `amp-inabox-host.js`, while Travis on commits run `gulp dist`, which builds `amp4ads-host-v0.js`. The test server does not actually run the `gulp serve` server (instead SauceLabs proxies port 8000 to the test server), so the change to `app.js` in #20154 is a no-op to the testing framework, and the inabox test would always ask for `amp-inabox-host.js`. That's why the inabox tests succeed on PR check, but fails on master.

I can make a new PR that makes these tests pass again, but is there a way to force a real master Travis check so we don't have to wait for it to make master red to know that it fails?

**Comment by lannka** [comment]
> so the change to app.js in #20154 is a no-op to the testing framework, and the inabox test would always ask for amp-inabox-host.js

I don't this this is correct, as many our integration tests rely on changes on app.js, for example the request-banks (defined in amp4test.js but installed in app.js).

The code that adds `app.js` to test server is here:
https://github.com/ampproject/amphtml/blob/master/build-system/tasks/karma.conf.js#L277

**Comment by lannka** [comment]
> is there a way to force a real master Travis check so we don't have to wait for it to make master red to know that it fails?

@rsimha do you have the answer?

**Comment by rsimha** [comment]
I believe you can do so on Travis via "more options -> trigger build"

**Comment by lannka** [comment]
@rsimha what @zombifier want is to trigger a full build for a pull request.

**Comment by rsimha** [comment]
Aha, there's no easy way to do so right now. We do `build` for PR builds and `dist` for push builds because `dist` takes a lot longer than `build` on Travis (~25 mins vs. ~5 mins), and we run way more push builds than PR builds on Travis.

It's part of our plans to unify the push builds and PR builds, but this is currently gated on significantly speeding up `dist` by defaulting to our single pass build, which runs a lot faster. Stay tuned for this.

Meanwhile, you can test locally by running:
```
gulp dist --fortesting
gulp test --integration --compiled [--headless] [--files=<test_file>]
```

**Comment by lannka** [comment]
to add, if the test is only failing on saucelabs, you need to [get saucelabs setup](https://github.com/ampproject/amphtml/blob/master/contributing/TESTING.md#testing-on-sauce-labs)

**Comment by zombifier** [comment]
I'll go with setting up Sauce Labs, since I still can't see anything wrong on local testing. Thanks.

**Comment by rsimha** [comment]
Thanks @zombifier, and sorry about the extra work. Let us know if you run into any issues.

**Comment by zombifier** [comment]
After some tinkering around on Sauce Labs I think my original assessment of the wrong port being used is correct. `app.js` is indeed being used, but the port that's used during testing is 9876, not 8000 (which is the port referenced in the inabox bootstrap script). The remote testing browsers can't reach the host script if `gulp serve` is not ran, which starts a separate instance on port 8000.

#20381 fixes this, but I still can't explain the previous behavior of the faulty tests passing during the PR checks, so unfortunately I can't 100% guarantee that master won't be red again.

**Comment by lannka** [comment]
ah, that seems right. shouldn't hard code port.

**Comment by lannka** [comment]
PR check does not run tests on Saucelabs, so no tests run under safari. that might explain why it passed  at that time.

I assume now it passes on Saucelabs right?

**Comment by zombifier** [comment]
Have been testing for a while, it seems to.

**Comment by zombifier** [comment]
Slightly related, the AMP example creative in `amp4test.js` uses `cdn` serving mode instead of fetching from the local server. From looking at the code and manual testing it indeed seems to be testing files from the AMP CDN. Is this really the case, and if so should it be changed?

**Comment by lannka** [comment]
>  amp4test.js uses cdn serving mode instead of fetching from the local server

the mode is read from a env var, which can be configured as command line argument when you run the test:

https://github.com/ampproject/amphtml/blob/master/build-system/amp4test.js#L226

**Comment by lannka** [comment]
Fixed by #20381
